src/components/course/data/utils.jsx

```diff
@@ -452,38 +452,49 @@ export const getCouponCodesDisabledEnrollmentReasonType = ({
   return undefined;
 };
 
+/**
+ * Determines whether the learner's subscription license is applicable to the course being
+ * viewed, based on the enterprise catalogs associated with the learner's subscription license.
+ * @param {Object} args
+ * @param {array} args.catalogsWithCourse List of catalogs that will be checked against the license.
+ * @param {Object} args.subscriptionLicense Learner's subscription license.
+ * @returns {boolean} True if the learner's subscription license is applicable to the course being viewed.
+ */
+function determineLicenseApplicableToCourse({
+  catalogsWithCourse,
+  subscriptionLicense,
+}) {
+  if (!subscriptionLicense) {
+    return false;
+  }
+  return catalogsWithCourse.includes(
+    subscriptionLicense.subscriptionPlan.enterpriseCatalogUuid,
+  );
```

Comment:
Wouldn't we want a different message in this case from the `!subscriptionLicense` case? Like if they have a current, active license, but the catalog for it does not match any `catalogsWithCourse`, don't we want to show a message indicating that the course is not in a catalog for them?

Reply:
You are correct, we show the same `DISABLED_ENROLL_REASON_TYPES.SUBSCRIPTION_LICENSE_NOT_ASSIGNED` message for both cases of no license _and_ a license associated with a mismatched catalog (assumes the course is indeed actually included in a catalog, though) and had a similar question, as we don't seem to differentiate between these today.

I was thinking about raising the question to UX, tackled as a separate ticket to provide copy and/or update the related Figma depicting these disabled enrollment reasons to differentiate between having no license and having a license with a mismatching catalog.

This case is not explicitly depicted in the Figma or code today.

Reply:
@iloveagent57 [inform] I started a thread posing this question to the UX team in `#enterprise-ux` ([Slack](https://twou.slack.com/archives/C06LQ2P5NV7/p1736516557682019)).

Will file an additional ticket if UX agrees we should differentiate the messaging between these two cases.